lib/discourse_api/api/sso.rb

```diff
@@ -3,24 +3,13 @@ module DiscourseApi
   module API
     module SSO
       def sync_sso(params = {})
-        sso = DiscourseApi::SingleSignOn.new
-        sso.sso_secret = params[:sso_secret]
-        sso.name = params[:name]
-        sso.username = params[:username]
-        sso.email = params[:email]
-        sso.external_id = params[:external_id]
-        sso.suppress_welcome_message = params[:suppress_welcome_message] === true
-        sso.avatar_url = params[:avatar_url]
-        sso.profile_background_url = params[:profile_background_url]
-        sso.card_background_url = params[:card_background_url]
-        sso.bio = params[:bio]
-        sso.title = params[:title]
-        sso.avatar_force_update = params[:avatar_force_update] === true
-        sso.add_groups = params[:add_groups]
-        sso.remove_groups = params[:remove_groups]
-        params.keys.select { |key| key.to_s.start_with?("custom") }.each do |custom_key|
-          sso.custom_fields[custom_key] = params[custom_key]
-        end
+        sso_secret = params.delete(:sso_secret)
+        _sso_url = params.delete(:sso_url)
+        querystring = URI.encode_www_form(params)
+
+        sso = DiscourseApi::SingleSignOn.unsigned_parse(querystring, sso_secret)
+
```

Comment:
I will update unsigned_parse to take hash

Reply:
This way we don't have to add SSO params to the sync_sso method every time we add them to DiscourseApi::SingleSignOn::ACCESSORS